waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/VerifierApi.kt

```diff
@@ -260,46 +266,51 @@ fun Application.verfierApi() {
                     }
                 }
             }) {
-                logger.info { "POST verify/state" }
-                val sessionId = call.parameters["state"]
-                logger.info { "State: $sessionId" }
-                verificationUseCase.verify(sessionId, context.request.call.receiveParameters().toMap())
-                    .onSuccess {
-                        val session = verificationUseCase.getSession(sessionId!!)
-                        if (session.walletInitiatedAuthState != null) {
-                            val state = session.walletInitiatedAuthState
-                            val code = UUID().toString()
-                            context.respondRedirect("openid://?code=$code&state=$state")
-                        } else {
-                            call.respond(HttpStatusCode.OK, it)
-                        }
-                    }.onFailure {
-                        logger.debug(it) { "Verification failed ($it)" }
-                        var errorDescription = it.localizedMessage
-                        logger.error { "Error: $errorDescription" }
-                        if (sessionId != null) {
-                            val session = verificationUseCase.getSession(sessionId)
+                runCatching {
+                    logger.info { "POST verify/state" }
+                    val sessionId = call.parameters["state"]
```

Comment:
maybe worth replacing this with:
```kotlin
val sessionId = call.parameters.getOrFail("state")
```
this would allow to update `verificationUseCase.verify` to take a non-nullable `sessionId` parameter
which would allow to remove null-safety for `sessionId` when [fetching](https://github.com/walt-id/waltid-identity/blob/224d44e53cd2d47f40bb54d39e524c1d20948726/waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/oidc/VerificationUseCase.kt#L118) the session object and have smth. like:
```kotlin
val session = OIDCVerifierService.getSession(sessionId)
```
the verification call in this case would look like:
```kotlin
verificationUseCase.verify(sessionId, context.request.call.receiveParameters().toMap())
    .onSuccess {
        processVerificationSuccessResult(sessionId, it)
    }.onFailure {
        processVerificationFailureResult(it, sessionId)
    }.also {
        verificationUseCase.notifySubscribers(sessionId)
    }
```